substrate/frame/nomination-pools/src/mock.rs

```diff
@@ -59,6 +60,19 @@ impl StakingMock {
 		x.insert(who, bonded);
 		BondedBalanceMap::set(&x)
 	}
+	/// Mimics a slash towards a pool specified by `pool_id`.
+	/// This reduces the bonded balance of a pool by `amount` and calls [`Pools::on_slash`] to
+	/// enact changes in the nomination-pool pallet.
+	///
+	/// Does not modify any [`SubPools::with_era`] of the pool as [`Default::default`] is passed for
+	/// `slashed_unlocking`.
+	pub fn slash_to(pool_id: PoolId, amount: Balance) {
```

Comment:
I think it would be cleaner to combine `set_bonded_balance` and `slash_to` in one. They are doing pretty much same thing except the extra `Pools::on_slash`. You can check if the new balance in `set_bonded_balance` is reducing from before and call `Pools::on_slash` if so. Wdyt?

Reply:
Forcing `on_slash` on every `set_bonded_balance` that reduces the `bonded_balance` decreases the ability to write more fine grained tests. For example in some tests `set_bonded_balance` is used to simply modify the `bonded_value` to make sure that this leads to certain behaviour. e.g. to test if the pool can be joined. I think basic tests like this should remain simple.

I think I should have just done a better job at combining new `TVL` checks with previous test.
Please let me know if this changes your mind.